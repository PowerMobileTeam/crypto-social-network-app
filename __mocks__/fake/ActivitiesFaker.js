
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

