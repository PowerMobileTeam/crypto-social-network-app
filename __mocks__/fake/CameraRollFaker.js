
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

