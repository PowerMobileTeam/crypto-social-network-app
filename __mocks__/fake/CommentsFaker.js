
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

