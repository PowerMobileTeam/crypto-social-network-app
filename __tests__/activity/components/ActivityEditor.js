import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import ActionSheet from 'react-native-actionsheet';
import * as Progress from 'react-native-progress';

import ActivityEditor from '../../../src/newsfeed/activity/ActivityEditor';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';



describe('Activity editor component', () => {

  let user, comments, entity, screen, newsfeed, toggleEdit;
  beforeEach(() => {
    newsfeed = {
      list: {
        updateActivity: () => {
          return new Promise((r, rr) => {
            return value;
          });
        }
      }
    }

    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    toggleEdit = jest.fn();
    screen = shallow(
      <ActivityEditor entity={activityResponse.activities[0]} toggleEdit={toggleEdit} navigation={navigation} newsfeed={newsfeed}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a Text input', async () => {
    screen.update();

    expect(screen.find('TextInput')).toHaveLength(1);
  });

  it('should set the default value when init, call toggle and submit',async () => {

    screen.update()
    let instance = screen.instance();
    expect(instance.state.text).toBe('Message');
    const spy = jest.spyOn(instance.props.newsfeed.list, 'updateActivity');
    const render = screen.dive();
    render.find('Button').at(1).props().onPress();

    expect(spy).toHaveBeenCalled();
  });

  it('should set the default value when init, call toggle and cancel',async () => {

    screen.update()
    let instance = screen.instance();
    expect(instance.state.text).toBe('Message');
    const spy = jest.spyOn(instance.props, 'toggleEdit');
    const render = screen.dive();
    render.find('Button').at(0).props().onPress();
    expect(spy).toHaveBeenCalled();
  });



  it('should set the default value when init, call toggle and cancel',async () => {

    screen.update()
    let instance = screen.instance();
    expect(instance.state.text).toBe('Message');

    screen.find('TextInput').forEach(child => {
      child.simulate('changeText', 'data');
    });

    const spy = jest.spyOn(instance.props, 'toggleEdit');
    const render = screen.dive();
    render.find('Button').at(0).props().onPress();
    expect(spy).toHaveBeenCalled();
    expect(instance.state.text).toBe('data');
  });


});
