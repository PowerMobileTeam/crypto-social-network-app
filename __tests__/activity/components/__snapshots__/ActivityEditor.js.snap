// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity editor component renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={4}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "width": "100%",
      }
    }
    value="Message"
  />
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        Object {
          "padding": 8,
        },
      ]
    }
  >
    <Button
      onPress={[Function]}
      title="Cancel"
    />
    <Button
      backgroundColor="#4690D6"
      onPress={[Function]}
      title="Save"
    />
  </Component>
</Component>
`;
